corehq/motech/requests.py

```diff
@@ -71,24 +73,36 @@ def get_url(self, uri):
         return '/'.join((self.base_url.rstrip('/'), uri.lstrip('/')))
 
     def delete(self, uri, **kwargs):
+        method_func = requests.delete if self.session is None else self.session.delete
```

Comment:
Not a huge fan of this if its not often used a lot of instances. what about changing this to always use a session?

```python
class Requests:
    def __init__():
        self._session = None

    @property
    def session(self):
        if not self._session:
            self._session = Session()
         return self._session
```

That's abbreviated to give an idea. it wouldn't handle session ending though

Reply:
That's an interesting idea. Up until now, the Requests class has only been used for RESTful APIs that support basic auth. We haven't needed sessions. I wonder whether subclassing it would make sense, so that the way we're using it everywhere else isn't affected by these specific (annoying) requirements for a session, and the subclass can always use a session.

Reply:
On second thoughts, subclassing is unnecessary too. Went with just always using a session. ea1c683